src/control/lib/spdk/ctests/nvme_control_ut.c

```diff
@@ -298,7 +298,6 @@ static int
 teardown(void **state)
 {
 	clean_ret(test_ret);
-	free(test_ret);
```

Comment:
Sorry for all the churn on this one. I must not have communicated clearly. So now that clean_ret again cleans out the struct pointers without freeing the struct pointer itself, this change should revert in order to avoid leaking memory.

Similarly in the Go code, we need to call C.clean_ret (as you're already doing), then C.free(ret) to finally free up the struct itself.